sycl/include/sycl/detail/spirv.hpp

```diff
@@ -498,22 +498,44 @@ AtomicMax(multi_ptr<T, AddressSpace, IsDecorated> MPtr, memory_scope Scope,
 }
 
 // Native shuffles map directly to a shuffle intrinsic:
-// - The Intel SPIR-V extension natively supports all arithmetic types
+// - The Intel SPIR-V extension natively supports all arithmetic types.
+//   However, OpenCL extension natively supports float vectors,
+//   integer vectors, half scalar and double scalar.
+//   For double vectors we perform emulation with scalar version.
 // - The CUDA shfl intrinsics do not support vectors, and we use the _i32
 //   variants for all scalar types
 #ifndef __NVPTX__
+
+template <typename T>
+struct TypeIsProhibitedForShuffleEmulation
+    : bool_constant<std::is_same_v<vector_element_t<T>, double>> {};
+
+template <typename T>
+struct VecTypeIsProhibitedForShuffleEmulation
+    : bool_constant<
+          (detail::get_vec_size<T>::size > 1) &&
+          TypeIsProhibitedForShuffleEmulation<vector_element_t<T>>::value> {};
+
 template <typename T>
 using EnableIfNativeShuffle =
-    detail::enable_if_t<detail::is_arithmetic<T>::value, T>;
-#else
+    detail::enable_if_t<detail::is_arithmetic<T>::value &&
```

Comment:
`std::enable_if_t`, ```std::is_arithmetic<T>```?

Reply:
Although they have similar names, `detail::is_arithmetic` accepts `vec` types that I need.

`std::enable_if_t` probably could be used.